### ISSUE TITLE
pandas.CSVDataSet cannot write utf8 chars to a file

### ISSUE THREAD

**Issue by jan-kaufmann** [quantumblacklabs/kedro#772]
## Description
I am working with pandas.CSVDataSet files that contain text. When these DataFrames contain non-ascii character such as 'less-than-or-equal' : \u2264 kedro throws an exception upon writing the file to disk. 

The problem seems to be limited to windows.

## Steps to Reproduce
1. Define a simple csv file on windows with a non-ascii character
```
column1,column2,column3
a,b,c
≤,2,3
```

2. Try to load and write that file in catalog.yml
```
test_read_file:
  type: pandas.CSVDataSet
  filepath: data/01_raw/t.csv
  load_args:
    encoding: utf8

test_write_file:
  type: pandas.CSVDataSet
  filepath: data/01_raw/t2.csv
  save_args:
    encoding: utf8
```

3. Define a simple node
```
def t_node(df):
    print(df)
    df.to_csv("this_works.csv", encoding="utf8") 
    return df

node(
func = t_node,
inputs = "test_read_file",
outputs = "test_write_file",
name = "t_node",
)
```

## Expected Result
The returned data frame should be written to t2.csv. 

## Actual Result
The plain pandas to_csv writes the file successful to disk. 
But kedro raises an exception:

> Failed while saving data to data set CSVDataSet(filepath=C:/Users/[user]/Documents/kedro/health_ai/data/01_raw/t2.csv, load_args={}, protocol=file, save_args={'encoding': utf8, 'index': False}).
'charmap' codec can't encode character '\u2264' in position 0: character maps to <undefined>

## Your Environment
Include as many relevant details about the environment in which you experienced the bug:

* Windows 10
*  Kedro 0.17.3
* Python 3.8.10
* Pandas 1.2.4

**Comment by deepyaman** [comment]
Hi @jan-kaufmann! Because of the way Kedro relies on `fsspec` under the hood, encoding has to be passed as `fs_args`. You can see an example in https://github.com/quantumblacklabs/kedro/issues/359#issuecomment-632760205.

**Comment by jan-kaufmann** [comment]
Hi @deepyaman,
first of all thanks for pointing me in the right direction.
```
  fs_args:
    open_args_save:
      encoding: utf8
```
allowed me to run the code!

However, I noticed the following things in the kedro documentation:
* CSVDataSet claims that ```"It uses pandas to handle the CSV file."``` Since utf-8 is panda's default encoding, why does kedro not use it?
* It also claims that I can configure it via ```save_args – Pandas options for saving CSV files.``` Since 'encoding' is one of pandas csv options I would expect that I can specify the encoding here aswell. Why does that not work?

To me it looks like kedro is ignoring parts of pandas/python here and there. I encountered a similar problem earlier when I wanted to pass a 'None' to pandas but in the yml file I had to leave the setting empty or pass 'NULL' instead. 

Is it a bug that the 'encoding' is not used from pandas or does it work as designed? In that case the documentation should make that more clear.

**Comment by lorenabalan** [comment]
Hi @jan-kaufmann , as @deepyaman mentioned the discrepancy is because of the `fsspec` wrap kedro has on top of the `pandas` call, i.e. https://github.com/quantumblacklabs/kedro/blob/master/kedro/extras/datasets/pandas/csv_dataset.py#L157-L158 

So it's not that kedro is purposefully ignoring anything, but it's the interaction between pandas and file handles / fsspec. More recently pandas has integrated with fsspec themselves, so some of these wrappers can go away, with a view to simplify the kedro API an remove the need for `open_args_save` / `open_args_load`. This has been improved on `develop` (see https://github.com/quantumblacklabs/kedro/commit/4f5f9c17bb8daf36a57048982c0b40d7a21fd3a0) and the change will be made available in the next breaking release, 0.18.0.

**TL;DR:** this has been fixed on `develop` and will be out in 0.18.0. 🙂

**Comment by lorenabalan** [comment]
Closing this as resolved.
